src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdSkinHelper.java

```diff
@@ -0,0 +1,105 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.helpers;
+
+import com.wikia.webdriver.common.contentpatterns.AdsContent;
+import com.wikia.webdriver.common.core.imageutilities.ImageComparison;
+import com.wikia.webdriver.common.core.imageutilities.ImageEditor;
+import com.wikia.webdriver.common.core.imageutilities.Shooter;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.Point;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+
+/**
+ * @author drets
+ * @ownership AdEng
+ */
+public class AdSkinHelper {
+
+  public static final String WIKIA_MESSAGE_BUBLE = "#WikiaNotifications div[id*='msg']";
+  public static final int IMAGES_THRESHOLD_PERCENT = 12;
+  private final WebDriver driver;
+  private final AdsComparison adsComparison;
+  private final String pathToLeftPart;
+  private final String pathToRightPart;
+  private final ImageEditor imageEditor;
+  private final ImageComparison imageComparison;
+  private final Shooter shooter;
+  private final int viewPortWidth;
+  private final int articleRightSideX;
+  private final int articleLeftSideX;
+  private final int startSkinY;
+
+
+  public AdSkinHelper(String pathToLeftPart, String pathToRightPart, WebDriver driver) {
+    this.pathToLeftPart = pathToLeftPart;
+    this.pathToRightPart = pathToRightPart;
+    this.driver = driver;
+    this.adsComparison = new AdsComparison();
+    this.imageEditor = new ImageEditor();
+    this.shooter = new Shooter();
+    this.imageComparison = new ImageComparison();
+    WebElement wikiaArticle = driver.findElement(By.id("WikiaPageBackground"));
+    WebElement globalNavigation = driver.findElement(By.id("globalNavigation"));
+    this.viewPortWidth = globalNavigation.getSize().getWidth();
+    this.startSkinY = globalNavigation.getSize().getHeight();
```

Comment:
Here is a lot of staff happening in the constructor, maybe move it to init method, or a lazy loading? There is a situation when, a can add a filed in object and it will throw an exception during construction